day_10/task_1/pipe_maze.cpp

```diff
@@ -14,3 +17,87 @@ MazePoint find_starting_point(const PipeMaze& pipe_maze)
     }
     return MazePoint(-1, -1);
 }
+
+bool is_connected_to_north(const char candi_south_pipe)
+{
+    static const std::string pipes_to_north{"|LJ"};
+    return pipes_to_north.contains(candi_south_pipe);
+}
+
+bool is_connected_to_south(const char candi_north_pipe)
+{
+    static const std::string pipes_to_south{"|F7"};
```

Comment:
Q: Is 'pipes to south ' always looks the same?

Reply:
A: Yes, it is always the same by design of the rules of the task. We have limited set of all kind of pipes and `pipes_to_south` is a subset of pipe kinds, which has at least one end pointed to south.